CMakeLists.txt

```diff
@@ -95,3 +95,74 @@ function(trusted_firmware_build)
   endif()
 
 endfunction()
+
+# Gets a list of commands to be performed after a successful Zephyr build that
+# involves trusted-firmware-m.
+function(trusted_firmware_get_post_build_commands out_commands)
+
+  if(NOT CONFIG_TFM_BL2_FALSE)
+
+    # Set default image versions if not defined elsewhere
+    if(NOT DEFINED TFM_IMAGE_VERSION_S)
+      set(TFM_IMAGE_VERSION_S 0.0.0+0)
+    endif()
+    if(NOT DEFINED TFM_IMAGE_VERSION_NS)
+      set(TFM_IMAGE_VERSION_NS 0.0.0+0)
+    endif()
+
+    set(TFM_MCUBOOT_DIR "${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/bl2/ext/mcuboot")
+    set(PREPROCESSED_FILE "${CMAKE_BINARY_DIR}/tfm/image_macros_preprocessed")
+
+    if(NOT DEFINED TFM_PUBLIC_KEY_FORMAT)
+      set(TFM_PUBLIC_KEY_FORMAT "full")
+    endif()
+
+    set(${out_commands}
+
+      # Sign secure binary image with public key
+      COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/imgtool.py
+        sign
+        --layout ${PREPROCESSED_FILE}_s.c
+        -k ${CONFIG_TFM_KEY_FILE_S}
+        --public-key-format ${TFM_PUBLIC_KEY_FORMAT}
+        --align 1
+        -v ${TFM_IMAGE_VERSION_S}
+        ${ADD_NS_IMAGE_MIN_VER}
+        ${ADD_SECURITY_COUNTER_S}
+        -H 0x400
+        ${CMAKE_BINARY_DIR}/tfm/install/outputs/${TFM_TARGET_PLATFORM}/tfm_s.bin
+        ${CMAKE_BINARY_DIR}/tfm_s_signed.bin
+
+      # Sign non-secure binary image with public key
+      COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/imgtool.py
+        sign
+        --layout ${PREPROCESSED_FILE}_ns.c
+        -k ${CONFIG_TFM_KEY_FILE_NS}
+        --public-key-format ${TFM_PUBLIC_KEY_FORMAT}
+        --align 1
+        -v ${TFM_IMAGE_VERSION_NS}
+        ${ADD_S_IMAGE_MIN_VER}
+        ${ADD_SECURITY_COUNTER_NS}
+        -H 0x400
+        --included-header
+        ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_BIN_NAME}
+        ${CMAKE_BINARY_DIR}/zephyr_ns_signed.bin
+
+      # Create concatenated binary image from the two independently signed binary files
+      COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/assemble.py
+        --layout ${PREPROCESSED_FILE}_s.c
+        -s ${CMAKE_BINARY_DIR}/tfm_s_signed.bin
+        -n ${CMAKE_BINARY_DIR}/zephyr_ns_signed.bin
+        -o ${CMAKE_BINARY_DIR}/tfm_sign.bin
+
+      # Copy mcuboot.bin
+      COMMAND ${CMAKE_COMMAND} -E copy
+        ${CMAKE_BINARY_DIR}/tfm/install/outputs/${TFM_TARGET_PLATFORM}/mcuboot.bin
```

Comment:
The `TFM_TARGET_PLATFORM` variable may not be set, that depends on current scope of the caller.
This may lead to unexpected behavior depending on the location from where this function is called.
Same comment applies to L133.

Probably this should be an argument to the function.

Reply:
This variable is supposed to be set in the respective board definition, for example in [mps2_an521/board.cmake](https://github.com/zephyrproject-rtos/zephyr/blob/ec9080a03610acd7105d7721432a5ed22c75d98c/boards/arm/mps2_an521/board.cmake#L4):
```
set(TFM_TARGET_PLATFORM "AN521")
```

How about adding here a check like:
```
    if(NOT DEFINED TFM_TARGET_PLATFORM)
      message(FATAL_ERROR "TFM_TARGET_PLATFORM is not set")
    endif()
```

Reply:
We are having CMake cached variables and `CONFIG_<name>` Kconfig variables available as globals.
If we start having other variables semi globally available throughout modules, then things becomes much harder to keep track of.

So unless this is Kconfig setting, I would like it to be passed by the caller.